ext/rinku/autolink.c

```diff
@@ -27,22 +27,21 @@
 #define strncasecmp	_strnicmp
 #endif
 
+// using supported protocol ?
 bool
 autolink_issafe(const uint8_t *link, size_t link_len)
 {
 	static const size_t valid_uris_count = 5;
 	static const char *valid_uris[] = {
-		"/", "http://", "https://", "ftp://", "mailto:"
+		"http://", "https://", "ftp://", "mailto:"
 	};
 
 	size_t i;
 
 	for (i = 0; i < valid_uris_count; ++i) {
 		size_t len = strlen(valid_uris[i]);
 
-		if (link_len > len &&
-			strncasecmp((char *)link, valid_uris[i], len) == 0 &&
-			rinku_isalnum(link[len]))
```

Comment:
just because there is 1 alphanum character does not mean it's safe to link ... implementing partial checks in multiple places seems like a bad security model ... ideally this method would be "is_safe_protocol" ...